docs/adr/adr-0028-making-our-component-apis-more-robust.md

```diff
@@ -0,0 +1,137 @@
+---
+kind: '📌 Architecture Decision Records'
+---
+# ADR 0028: Making our component APIs more robust
+
+<!-- TODO: change the date -->
+🗓️ 2023-03-.. · ✍️ Florian Sanders
+
+## Context
+
+Most of our components deal with async data.
+Our oldest components used to base their logic on whether properties handling data were set or not to determine whether to show a loader or data.
```

Comment:
nitpick: I find this sentence hard to read.

Reply:
You are right, I went for this instead:

```
Our oldest components had a simple logic:
- If their data properties were not `null`, they would display data.
- Otherwise, they would show a loader.
```